### PR TITLE
Replace Numpy arrays by torch tensors 

### PR DESCRIPTION
**Pull request by rajathpatel23**
## Description
Please include a summary of the change, the motivation, and any additionl context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

The PR changes all dependencies on numpy to torch tensors - resolves [ Issue 100](https://github.com/OpenMined/SyferText/issues/100)

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
Running the tutorial notebook and unit test
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: tests/local_mode/test_language.py [hunk 1]
```diff
@@ -38,8 +38,12 @@ def test_vector_valid_token_is_not_zero():
     """Test that the vector of a valid token is not all zeros"""
 
     doc = nlp("banana")
-    actual = doc[0].vector
-    zeros = np.zeros(actual.shape)
+    # actual = doc[0].vector
+    # Convert numpy vector from doc var to torch tensor
+    actual = torch.tensor(doc[0].vector, dtype=torch.float32)
```
Comment (Nilanshrajput) [inline tests/local_mode/test_language.py:43]: This conversion should not be required, the vector is expected to be torch.Tensor

File: tests/local_mode/test_language.py [hunk 2]
```diff
@@ -49,8 +53,12 @@ def test_vector_non_valid_token_is_zero():
     """Test that the vector of non valid token is all zeros"""
 
     doc = nlp("outofvocabularytoken")
-    actual = doc[0].vector
-    zeros = np.zeros(actual.shape)
+    # Convert numpy vector from doc var to torch tensor
```
Comment (Nilanshrajput) [inline tests/local_mode/test_language.py:56]: same as above

**Comment by Nilanshrajput** [comment]
@rajathpatel23  great work, 
regarding my comments: the doc[x].vector call should return a torch.tensor, not np array, for this take a look at vectors.py __getitem__() method, there you should explicitly convert that vector to tensor.

File: tests/local_mode/test_language.py (hunk 2 above)
Comment (rajathpatel23) [reply tests/local_mode/test_language.py:56]: Thanks Nilansh, I did remove the conversion for the vector coming from doc object, but to specify zeros tensor I wanted to know do we need to specify or know the data type of doc vector or is it going to be float32 ?

Comment (rajathpatel23) [reply tests/local_mode/test_language.py:56]: Also, from your commend below I understand that I can specify the datatype within the getitem function in vectors.py (where I am supposed to convert it) to resolve this, does this work ?

Comment (Nilanshrajput) [reply tests/local_mode/test_language.py:56]: check vectors.py, the vector loaded there is an np array, but you need to convert it to torch tensor, check __getitem__ method in vectors.py file,

**Comment by rajathpatel23** [comment]
Hi @Nilanshrajput  I have made changes by updating the conversion of vector to tensor in vectors.py and ran black format test

**Comment by rajathpatel23** [comment]
Let me if you want me to close this PR and create a new one with the latest commit

File: syfertext/vectors.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 import os
 from pathlib import Path
 import numpy as np
+import torch
 import importlib
 from typing import Union
```
Comment (Nilanshrajput) [inline syfertext/vectors.py:8]:
```suggestion
import syft as sy
hook = sy.TorchHook(torch)
```

File: syfertext/doc.py [hunk 4]
```diff
@@ -336,8 +338,11 @@ def get_token_vectors(self, excluded_tokens: Dict[str, Set[object]] = None) -> n
             if include_token:
```
Comment (Nilanshrajput) [inline syfertext/doc.py:338]: Since know the vectors are already tensors you don't need to explicitly convert them to torch tensor. Can you go through  doc.py and tokrn.py for such examples,(mostly go through functions which are related to vectors etc)

**Comment by Nilanshrajput** [comment]
@rajathpatel23  Can you investigate on errors encountered while running tests

File: syfertext/vectors.py (hunk 3 above)
Comment (AlanAboudib) [reply syfertext/vectors.py:8]: Why do we need the hooking here? It is already done in `__init__()`

File: syfertext/doc.py (hunk 4 above)
Comment (AlanAboudib) [reply syfertext/doc.py:338]: @rajathpatel23 would you please consider @Nilanshrajput comment here

File: syfertext/vectors.py (hunk 3 above)
Comment (Nilanshrajput) [reply syfertext/vectors.py:8]: Oh, I did not consider that, my bad.

Comment (AlanAboudib) [reply syfertext/vectors.py:8]: :+1:

File: syfertext/doc.py [hunk 5]
```diff
@@ -336,8 +338,11 @@ def get_token_vectors(self, excluded_tokens: Dict[str, Set[object]] = None) -> n
...
                 token_vectors.append(token.vector)
 
-        # Convert to Numpy array.
-        token_vectors = np.array(token_vectors)
+        # # Convert to Numpy array.
+        # token_vectors = np.array(token_vectors)
+
+        # Convert to torch tensor
+        token_vectors = torch.tensor(token_vectors, dtype=torch.float32)
```
Comment (AlanAboudib) [inline syfertext/doc.py:345]:
```suggestion
        # Convert to torch tensor
        token_vectors = torch.stack(tensors = token_vectors)
```

File: syfertext/doc.py [hunk 6]
```diff
@@ -338,11 +338,8 @@ def get_token_vectors(self, excluded_tokens: Dict[str, Set[object]] = None) -> t
             if include_token:
                 token_vectors.append(token.vector)
 
-        # # Convert to Numpy array.
-        # token_vectors = np.array(token_vectors)
-
         # Convert to torch tensor
-        token_vectors = torch.tensor(token_vectors, dtype=torch.float32)
+        token_vectors = torch.stack(tensors = token_vectors)
```
Comment (Nilanshrajput) [inline syfertext/doc.py:342]: @AlanAboudib  I was thinking maybe instead of appending to a list, just stack the tensors in the previous step? Probably will save some memory.

File: syfertext/doc.py (hunk 4 above)
Comment (rajathpatel23) [reply syfertext/doc.py:338]: @AlanAboudib sure, I am just confused with that function because the variable `token_vector` is list and was originally converted to numpy array due to which I decided to convert it to tensor because the return type of the function is changed to tensor from a numy array. However, let me it a try again today, review is more deeply. Also, I am using a Windows system due to which I am having problems configuring my environment to run tests locally which is the reason everything I commit is kind getting push to this PR.

**Comment by rajathpatel23** [comment]
@AlanAboudib , @Nilanshrajput  I am closing this request and will open a new request with all the updated and tested changes
